app/notification/model/notification_utils.py

```diff
@@ -0,0 +1,59 @@
+import re
+
+
+def convert_bool_value(data):
+    def convert_values(value):
+        if value is None:
+            return "Not provided"
+        elif value is True:
+            return "Yes"
+        elif value is False:
+            return "No"
+        else:
+            return value
+
+    if isinstance(data, list):
+        if all(isinstance(sublist, list) for sublist in data):
+            converted_data = [
+                [convert_values(value) for value in sublist]
+                for sublist in data
+            ]
+        else:
+            converted_data = [convert_values(value) for value in data]
+    else:
+        converted_data = convert_values(data)
+
+    return converted_data
```

Comment:
Can we add unit tests for this function to see inputs and expected outputs?

Also we could probably do this recursively:

```suggestion
def convert_list_to_boolean(lst):
    result = []
    for value in lst:
        if isinstance(value, list):
            result.append(convert_list_to_boolean(value))  # Recursive call
        else:
            if value is None:
                result.append("Not provided")
            elif value is True:
                result.append("Yes")
            elif value is False:
                result.append("No")
            else:
                result.append(value)
    return result
```